src/base/zonefile_fmt.rs

```diff
@@ -128,6 +142,59 @@ impl<W: fmt::Write> FormatWriter for SimpleWriter<W> {
     }
 }
 
+/// A simple writer that aligns some bits
+struct AlignedWriter<W> {
+    first: bool,
+    blocks: usize,
+    writer: W,
+}
+
+impl<W> AlignedWriter<W> {
+    fn new(writer: W) -> Self {
+        Self {
+            first: true,
+            blocks: 0,
+            writer,
+        }
+    }
+}
+
+impl<W: fmt::Write> FormatWriter for AlignedWriter<W> {
+    fn fmt_token(&mut self, args: fmt::Arguments<'_>) -> Result {
+        if !self.first {
+            let c = if self.blocks == 0 { '\t' } else { ' ' };
+            self.writer.write_char(c)?;
+        }
+        self.first = false;
+        self.writer.write_fmt(args)?;
+        Ok(())
+    }
+
+    fn begin_block(&mut self) -> Result {
+        self.blocks += 1;
+        Ok(())
+    }
+
+    fn end_block(&mut self) -> Result {
+        self.blocks -= 1;
+        Ok(())
+    }
+
+    fn fmt_comment(&mut self, _args: fmt::Arguments<'_>) -> Result {
+        Ok(())
+    }
+
+    fn newline(&mut self) -> Result {
+        self.writer.write_char('\n')?;
+        self.first = true;
+
+        // Little sanity thing, it _should_ already be 0
+        self.blocks = 0;
```

Comment:
Should this perhaps then be a `dbg_assert()` instead?

Reply:
Sure!